### PR TITLE
py-pycm: add new package with version 4.0

### PR DESCRIPTION
**Pull request by ChristopherChristofi**
PyPycm - 4.0: [release notes](https://github.com/sepandhaghighi/pycm/releases/tag/v4.0)

### CODE REVIEW

File: var/spack/repos/builtin/packages/py-pycm/package.py [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+    """Multi-class confusion matrix library in Python."""
+
+    homepage = "https://www.pycm.io"
+    pypi = "pycm/pycm-4.0.tar.gz"
+
+    license("MIT")
+
+    version("4.0", sha256="839e217eeb9a093be633ea746c5ca8b7ab6591d978762face892473c9f28959d")
+
+    depends_on("python", type=("build", "run"))
```
Comment (haampie) [inline var/spack/repos/builtin/packages/py-pycm/package.py]:
```suggestion
```

**Comment by ChristopherChristofi** [comment]
@spackbot run pipeline

**Comment by bernhardkaindl** [comment]
@ChristopherChristofi:

To be able to build "py-pycm: add new package with version 4.0" from https://github.com/spack/spack/pull/43251, I needed to add `depends_on("py-setuptools", type="build")` to it and `py-art`.

I copied the changes in this incremental PR from other packages, so I think it is fine to approve and merge this change onto your https://github.com/spack/spack/pull/43251 using this PR.

You can merge my tested changes by merging this PR into your branch (by pressing the approve and merge buttons):

https://github.com/kcl-eresearch/spack/pull/1 -  Add missing depends_on("py-setuptools", type="build") #1

**Review by bernhardkaindl — COMMENTED** [review]
@ChristopherChristofi:

To be able to build "py-pycm: add new package with version 4.0" from https://github.com/spack/spack/pull/43251, I needed to add `depends_on("py-setuptools", type="build")` to it and `py-art`.

I copied the changes in this incremental PR from other packages, so I think it is fine to approve and merge this change onto your https://github.com/spack/spack/pull/43251 using this PR.

You can merge my tested changes by merging this PR into your branch (by pressing the approve and merge buttons):

https://github.com/kcl-eresearch/spack/pull/1 -  Add missing depends_on("py-setuptools", type="build")

**Comment by ChristopherChristofi** [comment]
Hello @bernhardkaindl, not sure how I missed it, that is great, thank you for both spotting this and making the suggested updates

**Review by bernhardkaindl — APPROVED** [review]
I've tested building the new package, looks good to me now.

**Pull request merged**